Library/Homebrew/test/utils/curl_spec.rb

```diff
@@ -4,6 +4,115 @@
 require "utils/curl"
 
 describe "Utils::Curl" do
+  let(:details) {
+    details = {
+      normal:     {},
+      cloudflare: {},
+      incapsula:  {},
+    }
+
+    details[:normal][:no_cookie] = {
+      url:            "https://www.example.com/",
+      final_url:      nil,
+      status:         "403",
+      headers:        {
+        "age"            => "123456",
+        "cache-control"  => "max-age=604800",
+        "content-type"   => "text/html; charset=UTF-8",
+        "date"           => "Wed, 1 Jan 2020 01:23:45 GMT",
+        "etag"           => "\"3147526947+ident\"",
+        "expires"        => "Wed, 31 Jan 2020 01:23:45 GMT",
+        "last-modified"  => "Wed, 1 Jan 2020 00:00:00 GMT",
+        "server"         => "ECS (dcb/7EA2)",
+        "vary"           => "Accept-Encoding",
+        "x-cache"        => "HIT",
+        "content-length" => "3",
+      },
+      etag:           "3147526947+ident",
+      content_length: "3",
+      file:           "...",
+      file_hash:      nil,
+    }
+
+    details[:normal][:ok] = Marshal.load(Marshal.dump(details[:normal][:no_cookie]))
```

Comment:
One thing I meant to ask is whether we have a preferred way of creating a deep copy of a hash (as modifying a hash created using `#dup` or `#clone` can modify the values of the original hash).

The `Marshall.load(Marshall.dump(...))` approach is straightforward but we use ActiveSupport, so we could also load `#deep_dup` (i.e., `require "active_support/core_ext/object/deep_dup"` and then it's as simple as `details[:normal][:no_cookie].deep_dup`). Thoughts on this?

Reply:
Either seems fine. Unless it's causing bugs, though, I wouldn't worry about modifying the original hash, particular in a spec?

Reply:
Long story short, doing a deep copy is necessary for these tests to work correctly (and predictably). If we only did a shallow copy, changes to the copy (e.g., adding/modifying a header value) will also apply to the original variable and this will also affect other variables that copy the same original variable. This breaks tests but also makes the test data unpredictable (i.e., it would be different than what we expect).

That said, I think some of the changes discussed above will help to simplify the test data (can be addressed in the follow-up PR), so this may just be a short-term consideration. Either way, you've answered my question 😄